torch/testing/_internal/common_utils.py

```diff
@@ -1039,6 +1039,13 @@ def _compareScalars(self, a, b, *,
 
         return _compare_scalars_internal(a, b, rtol=rtol, atol=atol, equal_nan=equal_nan)
 
+    # Construct assert messages basd on internal messages and user message.
+    def _get_assert_msg(self, msg, debug_msg=None):
+        if msg is None:
+            return f"internal assert message: {debug_msg}"
+        else:
+            return f"user assert message: {msg}; internal assert message: {debug_msg}"
```

Comment:
What happens if msg and debug_msg are both None?

I'm not in love with all the string all being on the same line. What would just doing 

```
{debug_msg} \n {msg} 
```

look like when both {debug_msg} and {msg} are not None?

Reply:
sounds good. I actually had the same question. splitting into multiple lines does sound better. 
Also if both msg & debug_msg are None there are still the default line that's printed:
```
AssertionError: False is not true
```

Reply:
But if both are None doesn't the first branch `if msg is None` trigger and the string "internal assert message: None" get returned?

Reply:
Sounds like there should be a unit test for this :P

Reply:
We have test_testing.py for this purpose:

https://github.com/pytorch/pytorch/blob/71cfb73755f6e978828683a54468ce48fb44d44f/test/test_testing.py#L10

Reply:
hmm. good point on the double None scenario. since `AssertionError: False is not true` is always printed. would it be good to prepend the `debug_msg` with something like `Reason: {debug_msg}` ? this way at least to end-user it would look like 
```
AssertionError: False is not true
Reason: None
```

Reply:
In the case of both being None we should probably just return None, which I think would just print:

```
AssertionError: False is not true
```